app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -150,8 +153,35 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
 
     @Override
     public void onLongPress(MotionEvent e) {
-        // Also switch on long-press for accessibility
-        setMainImage(true, true);
+        openCurrentMainImageInGallery();
+    }
+
+    private void openCurrentMainImageInGallery() {
+        ImageType wantedImageType = imageTypes.get(mainImageIndex);
+
+        File file = null;
+        if (wantedImageType == ImageType.IMAGE_FRONT) {
+            file = Utils.retrieveCardImageAsFile(this, loyaltyCardId, ImageLocationType.front);
+        }
+        else if (wantedImageType == ImageType.IMAGE_BACK) {
+            file = Utils.retrieveCardImageAsFile(this, loyaltyCardId, ImageLocationType.back);
+        }
+
+        // Entry is either the barcode (not an image) or image retrieval failed
```

Comment:
I'm sorry for changing my mind but actually, let's:

```suggestion
        // Image retrieval failed
```

And start the if/else with ImageType.BARCODE and show a toast like "Only images can be opened in the gallery app" and return. That way users can't get confused why nothing happens when they long-press the barcode.